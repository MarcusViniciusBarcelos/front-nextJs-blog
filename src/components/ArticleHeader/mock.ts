import { ArticleHeaderProps } from '.';

// Mock dos dados fictícios
const mockArticleHeaderData: ArticleHeaderProps = {
  id: '1',
  title: 'Título do Artigo',
  excerpt: 'Excerto do artigo...',
  cover: {
    id: '1',
    alternativeText: 'Texto alternativo da capa',
    url: 'https://napratica.org.br/wp-content/uploads/2018/08/tecnica-feynman.jpg',
  },
  author: {
    id: '1',
    displayName: 'Autor do Artigo',
    slug: 'autor-do-artigo',
  },
  categories: [
    {
      id: '1',
      displayName: 'Categoria do Artigo',
      slug: 'categoria-do-artigo',
    },
  ],
  createdAt: '2022-02-18',
};

export default mockArticleHeaderData;
