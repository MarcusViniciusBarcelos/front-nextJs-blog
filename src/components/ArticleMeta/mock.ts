import { ArticleMetaProps } from '.';
import { data } from '../../api/dados.json';

const postData = data.posts.data[1];

const mockArticleMetaData: ArticleMetaProps = {
  createdAt: postData.attributes.createdAt,
  author: {
    id: postData.attributes.author.data.id,
    attributes: {
      slug: postData.attributes.author.data.attributes.slug,
      displayName: postData.attributes.author.data.attributes.displayName,
    },
  },
  categories: postData.attributes.categories.data.map((category) => ({
    id: category.id,
    attributes: {
      slug: category.attributes.slug,
      displayName: category.attributes.displayName,
    },
  })),
};

export default mockArticleMetaData;
