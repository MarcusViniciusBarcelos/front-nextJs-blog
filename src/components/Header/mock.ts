import { HeaderProps } from '.';

export default {
  blogName: 'Marcus Barcelos',
  blogDescription:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud.',
  logo: 'https://res.cloudinary.com/dm5kvicsz/image/upload/v1708022537/Whats_App_Image_2023_07_13_at_15_32_56_000d5b42b2.jpg',
  showText: true,
} as HeaderProps;
