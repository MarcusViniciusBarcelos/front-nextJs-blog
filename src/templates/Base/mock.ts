import { BaseTemplateProps } from '.';
import { data } from '../../api/dados.json';

const baseData = data.setting.data.attributes;
const logoData = baseData.logo.data;

const mockTemplateData: BaseTemplateProps = {
  settings: {
    id: data.setting.data.id,
    attributes: {
      blogName: baseData.blogName,
      logo: {
        data: {
          attributes: {
            alternativeText: logoData.attributes.alternativeText,
            url: logoData.attributes.url,
          },
          id: logoData.id,
        },
      },
      blogDescription: baseData.blogDescription,
      menuLink: baseData.menuLink.map((link) => ({
        id: link.id,
        link: link.link,
        newTab: link.newTab,
        text: link.text,
      })),
      footer: baseData.footer,
    },
  },
  children: 'ola mundo',
};

export default mockTemplateData;
