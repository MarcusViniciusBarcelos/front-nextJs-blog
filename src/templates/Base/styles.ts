import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    svg.search-ok-icon {
      width: 3rem;
      height: 3rem;
      margin-left: 0.5rem;
    }

    svg.search-cancel-icon {
      width: 3rem;
      height: 3rem;
      margin-left: 0.5rem;
      color: ${theme.colors.secondary};
    }
  `}
`;

export const HeaderContainer = styled.div`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.xhuge};
    padding: 0 ${theme.spacings.large};
  `}
`;

export const SearchContainer = styled.div`
  ${({ theme }) => css`
    margin: 0 auto;
    margin-bottom: ${theme.spacings.xlarge};
    padding: 0 ${theme.spacings.large};
    max-width: 120rem;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
  `}
`;

export const SearchInput = styled.input`
  ${({ theme }) => css`
    padding: 0.5rem ${theme.spacings.small};

    &:disabled {
      outline: 0.1rem solid ${theme.colors.secondary};
    }
  `}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    margin-bottom: ${theme.spacings.large};
  `}
`;

export const FooterContainer = styled.div`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    padding: ${theme.spacings.large};
  `}
`;
