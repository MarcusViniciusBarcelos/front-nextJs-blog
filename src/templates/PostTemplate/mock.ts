import { PostTemplateProps } from '.';
import { data } from '../../api/dados.json';

const baseData = data.setting.data.attributes;
const logoData = baseData.logo.data;
const postData = data.posts.data[2];

const mockTemplateData: PostTemplateProps = {
  settings: {
    id: data.setting.data.id,
    attributes: {
      blogName: baseData.blogName,
      logo: {
        data: {
          attributes: {
            alternativeText: logoData.attributes.alternativeText,
            url: logoData.attributes.url,
          },
          id: logoData.id,
        },
      },
      blogDescription: baseData.blogDescription,
      menuLink: baseData.menuLink.map((link) => ({
        id: link.id,
        link: link.link,
        newTab: link.newTab,
        text: link.text,
      })),
      footer: baseData.footer,
    },
  },
  post: {
    id: postData.id,
    createdAt: postData.attributes.createdAt,
    title: postData.attributes.title,
    content: postData.attributes.content,
    cover: {
      id: postData.attributes.cover.data.id,
      attributes: {
        alternativeText:
          postData.attributes.cover.data.attributes.alternativeText,
        url: postData.attributes.cover.data.attributes.url,
      },
    },
    categories: postData.attributes.categories.data.map((category) => ({
      id: category.id,
      attributes: {
        displayName: category.attributes.displayName,
        slug: category.attributes.slug,
      },
    })),
    author: {
      id: postData.attributes.author.data.id,
      attributes: {
        displayName: postData.attributes.author.data.attributes.displayName,
        slug: postData.attributes.author.data.attributes.slug,
      },
    },
    excerpt: postData.attributes.excerpt,
    tags: postData.attributes.tags.data.map((tag) => ({
      id: tag.id,
      attributes: {
        displayName: tag.attributes.displayName,
        slug: tag.attributes.slug,
      },
    })),
  },
};

export default mockTemplateData;
