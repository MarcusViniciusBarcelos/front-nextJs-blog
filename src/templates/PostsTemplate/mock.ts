import { PostsTemplateProps } from '.';
import { data } from '../../api/dados.json';

const baseData = data.setting.data.attributes;
const logoData = baseData.logo.data;

const mockTemplateData: PostsTemplateProps = {
  settings: {
    id: data.setting.data.id,
    blogName: baseData.blogName,
    logo: {
      alternativeText: logoData.attributes.alternativeText,
      url: logoData.attributes.url,
      id: logoData.id,
    },
    blogDescription: baseData.blogDescription,
    menuLink: baseData.menuLink.map((link) => ({
      id: link.id,
      link: link.link,
      newTab: link.newTab,
      text: link.text,
    })),
    text: baseData.text[0].children[0].text,
  },
  posts: data.posts.data.map((post) => ({
    id: post.id,
    title: post.attributes.title,
    slug: post.attributes.slug,
    cover: {
      id: post.attributes.cover.data.id,
      alternativeText: post.attributes.cover.data.attributes.alternativeText,
      url: post.attributes.cover.data.attributes.url,
    },
    content: post.attributes.content,
    excerpt: post.attributes.excerpt,
    tags: post.attributes.tags.data.map((tag) => ({
      displayName: tag.attributes.displayName,
      slug: tag.attributes.slug,
      id: tag.id,
    })),
  })),
};

export default mockTemplateData;
