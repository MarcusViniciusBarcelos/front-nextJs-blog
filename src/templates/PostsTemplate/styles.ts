import styled, { css } from 'styled-components';

export const ButtonContainer = styled.div`
  ${({ theme }) => css`
    padding: 0 ${theme.spacings.large};
    margin: ${theme.spacings.large} 0;
    display: flex;
    justify-content: center;
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    background: ${theme.colors.primary};
    color: ${theme.colors.white};
    padding: ${theme.spacings.small} ${theme.spacings.large};
    border: none;
    cursor: pointer;
    transition: all 0.3s ease-in-out;

    &:disabled {
      background: ${theme.colors.darkerGray};
    }
  `}
`;
