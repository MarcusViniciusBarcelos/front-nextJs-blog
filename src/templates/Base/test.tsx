import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { BaseTemplate, BaseTemplateProps } from '.';

import mock from './mock';

const props: BaseTemplateProps = mock;

describe('<Base />', () => {
  it('should render', () => {
    renderTheme(<BaseTemplate {...props} />);

    expect(screen.getByRole('heading', { name: 'Oi' })).toBeInTheDocument();
  });
});
