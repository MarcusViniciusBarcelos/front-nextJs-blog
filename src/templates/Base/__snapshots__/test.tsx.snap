// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base /> should match snapshot 1`] = `
.c15 {
  text-align: center;
  border-top: 0.1rem solid #DDDDDD;
  padding: 2.4rem;
}

.c15 a {
  color: inherit;
  text-decoration: none;
}

.c16 {
  font-size: calc(1.6rem + 0.2rem);
  line-height: 1.5;
}

.c16 p {
  margin: 2.4rem 0;
}

.c16 a,
.c16 a:visited,
.c16 a:link {
  color: #dc143c;
  text-decoration: none;
  transition: all 0.3s ease-in-out;
}

.c16 a:hover {
  filter: brightness(0.8);
}

.c16 code {
  font-family: monospace;
  color: #dc143c;
  font-size: 1.6rem;
  background: #DDDDDD;
  padding: 0.2rem;
  margin: 0.2rem;
}

.c16 pre {
  background: #000000;
  padding: 2.4rem;
  font-family: monospace;
  color: #FFFFFF;
  margin: 2.4rem 0;
  overflow-x: auto;
  width: 100%;
  font-size: 1.6rem;
}

.c16 pre code {
  color: inherit;
  background: inherit;
}

.c16 img {
  max-width: 100%;
}

.c16 .image {
  background: #DDDDDD;
  line-height: 0;
  margin: 2.4rem 0;
}

.c16 .image figcaption {
  font-size: 1.6rem;
  padding: 1.6rem;
  text-align: center;
  line-height: 1.3;
}

.c16 .image-style-side {
  max-width: 50%;
  float: right;
  margin: 2.4rem 0;
}

.c16 hr {
  border: none;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c16 ul,
.c16 ol {
  margin: 2.4rem 4.0rem;
}

.c16 .table {
  width: 100%;
  overflow: hidden;
  overflow-x: auto;
  margin: 2.4rem 0;
}

.c16 table {
  width: 100%;
  border-collapse: collapse;
}

.c16 table td,
.c16 table th {
  padding: 1.6rem;
  border: 0.1rem solid #DDDDDD;
}

.c16 blockquote {
  border-left: 0.5rem solid #dc143c;
  color: #aaaaaa;
  filter: brightness(80%);
  padding-left: 2.4rem;
  font-style: italic;
  margin: 2.4rem;
}

.c17 {
  position: fixed;
  background: #000000;
  color: #FFFFFF;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 4rem;
  height: 4rem;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c6 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c12 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: none;
}

.c7 {
  display: flex;
  align-items: center;
  text-decoration: none;
  color: inherit;
}

.c7 >img {
  width: 15rem;
  height: 15rem;
  border-radius: 50%;
}

.c10 {
  display: flex;
  align-items: center;
  justify-content: center;
  padding-bottom: 6.4rem;
  width: 100%;
  max-width: 96rem;
  color: #333333;
  font-size: 1.6rem;
  margin: 0 auto;
}

.c10 .c5 {
  margin: 0 0 calc(1.6rem - 1rem);
}

.c11 {
  display: flex;
  flex-flow: column wrap;
  margin-left: 3.2rem;
  max-width: 48rem;
}

.c2 {
  background: #000000;
  padding: 3.2rem;
  display: flex;
  position: fixed;
  z-index: 1;
  width: 100%;
  max-width: 32rem;
  height: 100vh;
  top: 0;
  left: 0%;
  transition: all 0.3s ease-in-out;
  overflow-y: auto;
  left: -30rem;
  overflow-y: hidden;
}

.c3 {
  margin: auto;
  width: 100%;
}

.c4 .c5 {
  display: flex;
  justify-content: center;
  margin: 0;
  margin-bottom: 4.0rem;
}

.c4 .c5 img {
  border: 0.5rem solid #dc143c;
}

.c0 {
  position: fixed;
  top: 2.4rem;
  color: #FFFFFF;
  background-color: #000000;
  z-index: 2;
  width: 3rem;
  height: 3rem;
  left: 26rem;
  transition: all 0.3s ease-in-out;
  left: 1rem;
  color: #FFFFFF;
}

.c8 {
  display: block;
  color: #FFFFFF;
  text-decoration: none;
  margin-bottom: 1.6rem;
  font-size: 1.8rem;
  border-right: 0.5rem solid #000000;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  border-right: 0.5rem solid #dc143c;
  color: #dc143c;
}

.c9 {
  margin-top: 6.4rem;
}

.c13 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  margin-bottom: 6.4rem;
}

.c14 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width: 768px) {
  .c16 {
    font-size: 2rem;
  }

  .c16 .image-style-side {
    max-width: 100%;
    float: none;
    margin: 2.4rem;
  }
}

@media (max-width: 768px) {
  .c2 {
    left: -32rem;
  }
}

@media (max-width: 768px) {
  .c0 {
    left: -0.5rem;
  }
}

<div>
  <div
    class=""
  >
    <a
      aria-label="Open or close menu"
      class="c0"
      href="#"
      title="Open or close menu"
    >
      <svg
        aria-hidden="true"
        aria-label="Open menu"
        class="c1"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </a>
    <div
      aria-hidden="true"
      class="c2"
    >
      <nav
        class="c3"
      >
        <div
          class="c4"
        >
          <h1
            class="c5 c6"
          >
            <a
              href="/"
            >
              <a
                class="c7"
                target="_self"
              >
                <img
                  alt="meu blog"
                  src="https://res.cloudinary.com/dm5kvicsz/image/upload/v1708022644/react_9bb785764c.png"
                />
              </a>
            </a>
          </h1>
        </div>
        <a
          class="c8"
          href="#link1"
          target="_self"
        >
          link 1
        </a>
        <a
          class="c8"
          href="#link2"
          target="_self"
        >
          link 2
        </a>
      </nav>
    </div>
    <div
      class="c9"
    >
      <header
        class="c10"
      >
        <h1
          class="c5 c6"
        >
          <a
            href="/"
          >
            <a
              class="c7"
              target="_self"
            >
              <img
                alt="meu blog - blog de testes"
                src="https://res.cloudinary.com/dm5kvicsz/image/upload/v1708022644/react_9bb785764c.png"
              />
            </a>
          </a>
        </h1>
        <div
          class="c11"
        >
          <h2
            class="c5 c12"
          >
            meu blog
          </h2>
          <p>
            blog de testes
          </p>
        </div>
      </header>
    </div>
    <div
      class="c13"
    >
      ola mundo
    </div>
    <div
      class="c14"
    >
      <footer
        class="c15"
      >
        <div
          class="c16"
        >
          meu footer de boas na lagoa
        </div>
      </footer>
    </div>
    <a
      aria-label="Go to top"
      class="c17"
      href="#"
      title="Go to top"
    >
      <svg
        aria-hidden="true"
        class="c1"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
        />
      </svg>
    </a>
  </div>
</div>
`;
